app/controllers/admin/content_controller.rb

```diff
@@ -113,6 +113,22 @@ def autosave
     render :text => nil
   end
 
+  def merge
+    @article_1 = Article.find(params[:id])
+    # check to see that article with entered ID exists and isn't the same as the current article
+    if !Article.find_by_id(params[:merge_with]).nil? && (params[:merge_with] != params[:id])
+      @article_1.merge_with(params[:merge_with])
+      flash[:notice] = _('Article was successfully merged')
+      redirect_to :action => 'index'
+    elsif Article.find_by_id(params[:merge_with]).nil?
```

Comment:
So you're doing two `find_by_id`s here, which is going to make the same database query twice. I'd recommend storing the result of this above the conditional so it will only do this database query once.